### ISSUE TITLE
Duplicated CVE in 0.5.0 appears to disregard policy

### ISSUE THREAD

**Issue by jbouse** [anchore/anchore-engine#273]
**Is this a BUG REPORT or a FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Anchore Engine and Anchore CLI if applicable**:
```
# anchore-cli --version
anchore-cli, version 0.5.0

# anchore-cli system status
Service analyzer (d021a83d482c, http://engine-analyzer:8228): up
Service apiext (anchore-quickstart, http://engine-api:8228): up
Service policy_engine (anchore-quickstart, http://engine-policy-engine:8228): up
Service catalog (anchore-quickstart, http://engine-catalog:8228): up
Service analyzer (329a0209dd76, http://engine-analyzer:8228): up
Service analyzer (0f202c6b389c, http://engine-analyzer:8228): up
Service simplequeue (anchore-quickstart, http://engine-simpleq:8228): up

Engine DB Version: 0.0.11
Engine Code Version: 0.5.0
```

**What happened**:
Policy bundle provides the following rule:

```
...
                {
                    "action": "STOP",
                    "gate": "vulnerabilities",
                    "id": "d4b058ec-bbe8-4f8e-a54f-03ec0a0cd022",
                    "params": [
                        {
                            "name": "package_type",
                            "value": "all"
                        },
                        {
                            "name": "severity_comparison",
                            "value": ">="
                        },
                        {
                            "name": "severity",
                            "value": "high"
                        },
                        {
                            "name": "fix_available",
                            "value": "true"
                        }
                    ],
                    "trigger": "package"
                },
                {
                    "action": "WARN",
                    "gate": "vulnerabilities",
                    "id": "55a84fcb-3301-4ef8-b0e1-aeddf796024c",
                    "params": [
                        {
                            "name": "package_type",
                            "value": "all"
                        },
                        {
                            "name": "severity_comparison",
                            "value": ">="
                        },
                        {
                            "name": "severity",
                            "value": "high"
                        },
                        {
                            "name": "fix_available",
                            "value": "false"
                        }
                    ],
                    "trigger": "package"
                },
                {
                    "action": "STOP",
                    "gate": "vulnerabilities",
                    "id": "295adc9f-28b9-4aab-a5d6-833cf7727db7",
                    "params": [
                        {
                            "name": "package_type",
                            "value": "all"
                        },
                        {
                            "name": "severity_comparison",
                            "value": "="
                        },
                        {
                            "name": "severity",
                            "value": "critical"
                        }
                    ],
                    "trigger": "package"
                }
...
```

When running against one of our images that is affected by CVE-2017-16032 using v0.5.0 this policy fails with duplication of the CVE as:

```
vulnerabilities        package        HIGH Vulnerability found in os package type (dpkg) - libidn11 (CVE-2017-14062 - https://security-tracker.debian.org/tracker/CVE-2017-14062)             stop          
vulnerabilities        package        HIGH Vulnerability found in os package type (dpkg) - libidn11 (CVE-2017-14062 - https://security-tracker.debian.org/tracker/CVE-2017-14062)             warn          

```

**What did you expect to happen**:
When this same policy is used to test the same Docker image using v0.4.x the resulting policy is a pass without duplication:

```
vulnerabilities        package        HIGH Vulnerability found in os package type (dpkg) - libidn11 (CVE-2017-14062 - https://security-tracker.debian.org/tracker/CVE-2017-14062)             warn          
```

**What docker images are you using**:
Tested and failed with anchore/anchore-engine:v0.5.0
Tested and successed with anchore/anchore-engine:v0.4.1 and anchore/anchore-engine:v0.4.2

**How to reproduce the issue**:
Create a policy bundle with the above mentioned rule to WARN if CVE Severity >= HIGH and no fix available, STOP if CVE Severity >= HIGH and fix is available, STOP if CVE Severity = CRITICAL. Then test with an image with applicable CVE affecting it.

**Anything else we need to know**:
Deployed via docker-compose using external AWS Aurora PostgreSQL RDS 10.7 DB

**Comment by jbouse** [comment]
Not sure if [this ](https://github.com/anchore/anchore-engine/compare/v0.4.2...v0.5.0#diff-d05c84a11f814fbe5aea9ef0a4b3bbaeL81)might be a source of the issue but seems to be related change in code between 0.4.2 and 0.5.0

**Comment by zhill** [comment]
Is that the full set of vuln gates for the policy?

What feeds are synced in this installation?

**Comment by zhill** [comment]
What image is this against?

**Comment by zhill** [comment]
I'm able to reproduce the issue. Have a fix in testing.

**Comment by jbouse** [comment]
I've reproduced again under a clean situation... full [sterlingts-policy.json](https://gist.github.com/jbouse/34a3c1ceb45ae71629a6bcc48bd56786) bundle can be found here... Steps took to reproduce were as follows:

```
$ mkdir aetest
$ cd aetest/
$ docker pull docker.io/anchore/anchore-engine:latest
$ docker create --name ae docker.io/anchore/anchore-engine:latest
$ docker cp ae:/docker-compose.yaml ./docker-compose.yaml
$ docker rm ae
$ docker-compose pull
$ docker-compose up -d
$ docker-compose exec engine-api anchore-cli system wait
$ docker ps 
$ docker cp sterlingts-policy.json <engine-api container ID>:/tmp
$ docker-compose exec engine-api anchore-cli policy add /tmp/sterlingts-policy.json
$ docker-compose exec engine-api anchore-cli policy list
$ docker-compose exec engine-api anchore-cli policy activate 6719704e-aada-42ef-85c1-441aab4854a9
$ docker-compose exec engine-api anchore-cli policy list
$ docker-compose exec engine-api anchore-cli image add mcr.microsoft.com/dotnet/core/aspnet:2.1
$ docker-compose exec engine-api anchore-cli image wait mcr.microsoft.com/dotnet/core/aspnet:2.1
$ docker-compose exec engine-api anchore-cli evaluate check mcr.microsoft.com/dotnet/core/aspnet:2.1 --detail
```

**Comment by zhill** [comment]
Fixed by 5fea005c68195ce89415411d4c5e0349bd05cd5a

Present in 0.5 branch for 0.5.1 release. Will be merged to master as well.
